core/src/main/java/org/carbondata/core/carbon/datastore/block/TableBlockInfo.java

```diff
@@ -185,6 +207,16 @@ public long getBlockLength() {
         > ((TableBlockInfo) other).blockOffset + ((TableBlockInfo) other).blockLength) {
       return 1;
     }
+    //compare the startBlockLetNumber
+    int diffStartBlockLetNumber =
+        blockletInfos.getStartBlockletNumber() - ((TableBlockInfo) other).blockletInfos
```

Comment:
Hi ,

Would like to suggest a couple of things here remove Comparable<Distributable>  from the class definition as Distributable is already implementing it.

Let use generics so that we donot have to typeCast everywhere in the compare method for that need to do two things:
1) Change the class definition of Distributable  to :  
public abstract class Distributable<T extends Distributable<T>> implements Comparable<T>

2)  Change the compareTo method definition to:  
public int compareTo(TableBlockInfo other)

Let me know incase you are facing any isssue in doing this I will provide the patch if needed.